Unmet peer dependencies

### Describe the bug

```
ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

.
└─┬ vite-ssg 0.21.0
  └── ✕ unmet peer @vueuse/head@"^0.5.0 || ^0.6.0 || ^0.7.0": found 0.8.2
```

This line should be updated: https://github.com/antfu/vite-ssg/blob/590580488475cdee5454799373087105449b13c1/package.json#L86

### Reproduction

https://github.com/ansidev/blog/pull/87

### System Info

```Shell
.
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/README.md
+++ b/README.md
@@ -120,7 +120,7 @@ export default defineConfig({
   ssgOptions: {
     crittersOptions: {
       // E.g., change the preload strategy
-      preload: "media",
+      preload: 'media',
       // Other options: https://github.com/GoogleChromeLabs/critters#usage
     },
   },

--- a/package.json
+++ b/package.json
@@ -83,7 +83,7 @@
     "release": "bumpp && npm publish"
   },
   "peerDependencies": {
-    "@vueuse/head": "^0.5.0 || ^0.6.0 || ^0.7.0 || ^0.8.0 || ^0.9.0",
+    "@vueuse/head": "^1.0.0",
     "critters": "^0.0.16",
     "vite": "^2.0.0 || ^3.0.0-0",
     "vue": "^3.2.10",
@@ -98,6 +98,7 @@
     }
   },
   "dependencies": {
+    "@unhead/dom": "^1.0.2",
     "fs-extra": "^10.1.0",
     "html-minifier": "^4.0.0",
     "html5parser": "^2.0.2",
@@ -113,7 +114,7 @@
     "@types/jsdom": "^20.0.0",
     "@types/prettier": "^2.7.1",
     "@types/yargs": "^17.0.13",
-    "@vueuse/head": "^0.9.8",
+    "@vueuse/head": "^1.0.16",
     "bumpp": "^8.2.1",
     "critters": "^0.0.16",
     "eslint": "^8.26.0",

--- a/src/node/build.ts
+++ b/src/node/build.ts
@@ -11,6 +11,7 @@ import { nextTick } from 'vue'
 import { JSDOM } from 'jsdom'
 import type { VitePluginPWAAPI } from 'vite-plugin-pwa'
 import type { RouteRecordRaw } from 'vue-router'
+import { renderDOMHead } from '@unhead/dom'
 import type { ViteSSGContext, ViteSSGOptions } from '../types'
 import { serializeState } from '../utils/state'
 import { renderPreloadLinks } from './preload-links'
@@ -166,10 +167,8 @@ export async function build(ssgOptions: Partial<ViteSSGOptions> = {}, viteConfig
         renderPreloadLinks(jsdom.window.document, ctx.modules || new Set<string>(), ssrManifest)
 
         // render head
-        head?.updateDOM(jsdom.window.document)
-
-        // wait for waiting HTML changes to be applied [see https://github.com/antfu/vite-ssg/pull/312]
-        await nextTick()
+        if (head)
+          await renderDOMHead(head.unhead, { document: jsdom.window.document })
 
         const html = jsdom.serialize()
         let transformed = (await onPageRendered?.(route, html, appCtx)) || html

